src/components/map-controls.js

```diff
@@ -179,213 +109,88 @@ export default class MapControls extends PureComponent {
     if (this.props.isDragging) {
       return config.CURSOR.GRABBING;
     }
-    if (this.state.isHovering) {
+    if (this.props.isHovering) {
       return config.CURSOR.POINTER;
     }
     return config.CURSOR.GRAB;
   }
 
-  _updateViewport(opts) {
-    let viewport = Object.assign({
-      latitude: this.props.latitude,
-      longitude: this.props.longitude,
-      zoom: this.props.zoom,
-      bearing: this.props.bearing,
-      pitch: this.props.pitch,
-      altitude: this.props.altitude,
-      isDragging: this.props.isDragging,
-      startDragLngLat: this.props.startDragLngLat,
-      startBearing: this.props.startBearing,
-      startPitch: this.props.startPitch
-    }, opts);
-
-    viewport = this._applyConstraints(viewport);
-
-    // if (viewport.startDragLngLat) {
-    //   const dragViewport = new FlatMercatorViewport(Object.assign({}, this.props, {
-    //     longitude: viewport.startDragLngLat[0],
-    //     latitude: viewport.startDragLngLat[1]
-    //   }));
-    //   this.setState({dragViewport});
-    // }
-
-    return this.props.onChangeViewport(viewport);
+  _updateViewport(...opts) {
```

Comment:
Since there is no longer any references to `Viewports`, should we call this `_updateMapState` instead to complete the nice `MapState` abstraction here and avoid conflating concepts? 

For consistency, it would probably have been nice to change the callback name to `onChangeMapState` as well but that is not worth it.